### DISCUSSION TITLE
Generyczne tworzenie portów

### DISCUSSION THREAD

**Discussion by dmt1337x** [Lowgular/extension-documentation#6] [New Ideas]
Będąc w DTO/Context/Query/Command w Actions chciałbym mieć możliwość stworzenia abstrakcji. Nazwa generowałaby się automatycznie, podajemy tylko nazwę metody.

**Comment by MisiakLowgular** [comment]
I would make the definition of this feature request a bit more generic. When I'm in a file containing an entity of type `OBJECT` or `DATA_STRUCTURE` **AND** this entity has a relationship of type `PARAMETER` (and maybe also `RETURN`?) Lowgular Actions suggest creating the "parent" entity.

**Reply by MisiakLowgular** [reply]
for example what @dmt1337x suggested
```json
{
  "entityRelations": {
    "commandPort": {
      "command": "PARAMETER",
    },
  },
  "entities": {
    "command": {
      "id": "command",
      "type": "OBJECT",
      "suffix": "Command",
      "folder": "application/primary/command",
      "insideOf": "Lib"
    },
    "commandPort": {
      "id": "commandPort",
      "type": "ABSTRACTION",
      "suffix": "CommandPort",
      "folder": "application/primary/command",
      "insideOf": "Lib"
    },
  }
}
```
After creating `command` entity when I go to its file and bring up Lowgular actions there is an option like "Create associated entity with a method" (name needs refinement) and I can choose `commandPort` entity type
